src/profile-logic/profile-data.js

```diff
@@ -417,86 +417,97 @@ export function collapsePlatformStackFrames(thread: Thread): Thread {
   });
 }
 
+export function filterThreadToSearchStrings(
+  thread: Thread,
+  searchStrings: string[] | null
+): Thread {
+  return timeCode('filterThreadToSearchStrings', () => {
+    if (!searchStrings || !searchStrings.length) {
+      return thread;
+    }
+
+    return searchStrings.reduce(filterThreadToSearchString, thread);
+  });
+}
+
 export function filterThreadToSearchString(
   thread: Thread,
   searchString: string
 ): Thread {
-  return timeCode('filterThreadToSearchString', () => {
-    if (searchString === '') {
-      return thread;
+  if (!searchString) {
+    return thread;
+  }
+  const lowercaseSearchString = searchString.toLowerCase();
+  const {
+    samples,
+    funcTable,
+    frameTable,
+    stackTable,
+    stringTable,
+    resourceTable,
+  } = thread;
+
+  function computeFuncMatchesFilter(func) {
+    const nameIndex = funcTable.name[func];
+    const nameString = stringTable.getString(nameIndex);
+    if (nameString.toLowerCase().includes(lowercaseSearchString)) {
+      return true;
     }
-    const lowercaseSearchString = searchString.toLowerCase();
-    const {
-      samples,
-      funcTable,
-      frameTable,
-      stackTable,
-      stringTable,
-      resourceTable,
-    } = thread;
-
-    function computeFuncMatchesFilter(func) {
-      const nameIndex = funcTable.name[func];
-      const nameString = stringTable.getString(nameIndex);
-      if (nameString.toLowerCase().includes(lowercaseSearchString)) {
+
+    const fileNameIndex = funcTable.fileName[func];
+    if (fileNameIndex !== null) {
+      const fileNameString = stringTable.getString(fileNameIndex);
+      if (fileNameString.toLowerCase().includes(lowercaseSearchString)) {
         return true;
       }
+    }
 
-      const fileNameIndex = funcTable.fileName[func];
-      if (fileNameIndex !== null) {
-        const fileNameString = stringTable.getString(fileNameIndex);
-        if (fileNameString.toLowerCase().includes(lowercaseSearchString)) {
-          return true;
-        }
+    const resourceIndex = funcTable.resource[func];
+    const resourceNameIndex = resourceTable.name[resourceIndex];
+    if (resourceNameIndex !== undefined) {
+      const resourceNameString = stringTable.getString(resourceNameIndex);
+      if (resourceNameString.toLowerCase().includes(lowercaseSearchString)) {
```

Comment:
Optional observation: I wonder if this toLowerCase is expensive here.

Reply:
I don't think this is too bad... but if we ever have issues around it we should profile and see if this comes from here :)